### PR TITLE
Confirmation modal on Swap

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

CAN REVIEW but still WIP!

https://user-images.githubusercontent.com/31534717/135116611-ddd916d5-1d6f-4a99-9de9-1c5da6438672.mov

- This enhances the text and styling for the pending confirmation modals.
- It's aim is to highlight the steps involved to submit or cancel an order and to approve a token. Both for EOA's and Smart Contract wallets.
- The component in `src/custom/components/TransactionConfirmationModal/index.tsx` uses the `walletTypes` and `orderTypes` (both `enums`) to display the right content.
 
# PENDING/TODO:
- Right now the `orderType` is hardcoded (`src/custom/components/TransactionConfirmationModal/index.tsx` L375) and needs an incoming prop to get it working as intended.
- Need to trigger this modal for token approvals (src/custom/pages/Swap/SwapMod.tsx - L766) and have it pass the orderType (from `enum` src/custom/components/TransactionConfirmationModal/index.tsx - L345). Will need help with that to expedite this PR.

### CODE REVIEW

File: src/components/AccountDetails/index.tsx [hunk 1]
```diff
@@ -174,6 +174,8 @@ const IconWrapper = styled.div<{ size?: number }>`
   span {
     height: ${({ size }) => (size ? size + 'px' : '32px')};
     width: ${({ size }) => (size ? size + 'px' : '32px')};
+    height: 50px;
+    width: 50px;
```
Comment (biocom) [inline src/components/AccountDetails/index.tsx:178]: Need to move this.

**Comment by elena-zh** [comment]
Hey @biocom , currently, there is no waiting pop-up when we run approve/wrap/unwrap transactions in the expert mode. There is an enhancement for this on the board - #1137 . Could we address it in this PR?

**Review by alfetopito — COMMENTED** [review]
LGTM

~What's missing?~

Sorry, missed the PR description :facepalm: 

What part specifically do you need help with?

**Comment by elena-zh** [comment]
Tiny UI issue: 
wallet icon is is not in the real center of the circle in the mobile view:
image

**Comment by biocom** [comment]
> Tiny UI issue: wallet icon is is not in the real center of the circle in the mobile view: image

Baby fox??

**Comment by anxolin** [comment]
Merging to consolidate all the waterfalls. Reviews are still welcomed, the idea is:
- Capture the comments of pending issues in: https://docs.google.com/document/d/1_I0yPgocBqbKGnFLC1L3nvlS3nMXZcNLOPhvR8YAxi4/edit#   
- Aggregate all pending PRs into develop
- Solve all the issues :) 
- Finish the Gnosis Safe / Presign tasks: https://github.com/gnosis/cowswap/issues/1305    
- Do testing once all is fixed and finished. Note that these functionalities required too many PRs to implement in iterations. For QA probably is too many, this is why I want to close a release next week so we can test it all merged together.

**Pull request merged**